### ISSUE TITLE
The rules configured in gcp_posture.xml do not take effect

### ISSUE THREAD

**Issue by lchico** [wazuh/wazuh#24943]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.9.0-alpha3 | Wazuh component | Manager | Packages | AmazonLinux |

While testing the [Release 4.9.0 - Alpha 3 - E2E UX tests - GCP integration](https://github.com/wazuh/wazuh/issues/24858#top), I confirmed that the issue reported in https://github.com/wazuh/wazuh/issues/21538 is still present in this version.

To test it, I used the following configuration:

<details>
<summary>ossec.conf on the manager</summary>

```
  <gcp-pubsub>
    <pull_on_start>yes</pull_on_start>
    <interval>1m</interval>
    <project_id>wazuh-dev-xxxxxxxxx</project_id>
    <subscription_name>xxxxxxxx-sub</subscription_name>
    <credentials_file>wodles/gcloud/credentials.json</credentials_file>
  </gcp-pubsub>
```
</details>

I added the following to /var/ossec/etc/rules/gcp_posture.xml:

<details>
<summary>gcp_posture.xml</summary>

```
<group name="gcp,">
  <!-- Misconfiguration detection -->
    <rule id="100200" level="10">
        <if_sid>65000</if_sid>
        <field name="gcp.finding.findingClass">MISCONFIGURATION</field>
        <description>A $(gcp.finding.findingClass) with $(gcp.finding.severity) severity has been discovered on the GCP project $(gcp.resource.projectDisplayName). $(gcp.finding.description)</description>
        <mitre>
          <id>T1562</id>
        </mitre>
    </rule>

  <!-- Threat detection -->
    <rule id="100201" level="10">
        <if_sid>65000</if_sid>
        <field name="gcp.finding.findingClass">THREAT</field>
        <description>A $(gcp.finding.findingClass) with $(gcp.finding.severity) severity has been discovered on the GCP project $(gcp.resource.projectDisplayName). $(gcp.finding.category).</description>
        <mitre>
          <id>T1562</id>
        </mitre>
    </rule>
</group>
```
</details>

I then [followed these steps](https://documentation-dev.wazuh.com/v4.9.0-alpha3/cloud-security/gcp/posture-management.html#network-misconfigurations) to generate the alert.

<details>
<summary> Add new firewall rule: test-4-9-0-alpha3 </summary>

```log
** Alert 1721824114.161616: - gcp,
2024 Jul 24 12:28:34 wazuh-server->Wazuh-GCloud
Rule: 65042 (level 3) -> 'GCP notice event on project wazuh-dev-xxxxxxxxx, monitored resource type: gce_firewall_rule.'
{"integration": "gcp", "gcp": {"insertId":"5ty4yce19oyc","labels":{"compute.googleapis.com/root_trigger_id":"82cb4dae-b711-405b-b654-d7b9f3cae9ea"},"logName":"projects/wazuh-dev-xxxxxxxxx/logs/cloudaudit.googleapis.com%2Factivity","operation":{"id":"operation-1721824104805-61dfd6a21f4d4-28b878cc-ca1f6046","last":true,"producer":"compute.googleapis.com"},"protoPayload":{"@type":"type.googleapis.com/google.cloud.audit.AuditLog","authenticationInfo":{"principalEmail":"xxxxxxxxx@wazuh.com"},"methodName":"beta.compute.firewalls.insert","request":{"@type":"type.googleapis.com/compute.firewalls.insert"},"requestMetadata":{"callerIp":"2803:9800:9403:4699:fc18:6348:50cf:7679","callerSuppliedUserAgent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36,gzip(gfe),gzip(gfe)"},"resourceName":"projects/wazuh-dev-xxxxxxxxx/global/firewalls/test-4-9-0-alpha3","serviceName":"compute.googleapis.com"},"receiveTimestamp":"2024-07-24T12:28:30.304706533Z","resource":{"labels":{"firewall_rule_id":"4624613645442960774","project_id":"wazuh-dev-xxxxxxxxx"},"type":"gce_firewall_rule"},"severity":"NOTICE","timestamp":"2024-07-24T12:28:29.997471Z"}}
integration: gcp
gcp.insertId: 5ty4yce19oyc
gcp.labels.compute.googleapis.com/root_trigger_id: 82cb4dae-b711-405b-b654-d7b9f3cae9ea
gcp.logName: projects/wazuh-dev-xxxxxxxxx/logs/cloudaudit.googleapis.com%2Factivity
gcp.operation.id: operation-1721824104805-61dfd6a21f4d4-28b878cc-ca1f6046
gcp.operation.last: true
gcp.operation.producer: compute.googleapis.com
gcp.protoPayload.@type: type.googleapis.com/google.cloud.audit.AuditLog
gcp.protoPayload.authenticationInfo.principalEmail: xxxxxxxxx@wazuh.com
gcp.protoPayload.methodName: beta.compute.firewalls.insert
gcp.protoPayload.request.@type: type.googleapis.com/compute.firewalls.insert
gcp.protoPayload.requestMetadata.callerIp: 2803:9800:9403:4699:fc18:6348:50cf:7679
gcp.protoPayload.requestMetadata.callerSuppliedUserAgent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36,gzip(gfe),gzip(gfe)
gcp.protoPayload.resourceName: projects/wazuh-dev-xxxxxxxxx/global/firewalls/test-4-9-0-alpha3
gcp.protoPayload.serviceName: compute.googleapis.com
gcp.receiveTimestamp: 2024-07-24T12:28:30.304706533Z
gcp.resource.labels.firewall_rule_id: 4624613645442960774
gcp.resource.labels.project_id: wazuh-dev-xxxxxxxxx
gcp.resource.type: gce_firewall_rule
gcp.severity: NOTICE
gcp.timestamp: 2024-07-24T12:28:29.997471Z
```
<img src="https://github.com/user-attachments/assets/7940e6c8-9802-49c6-a78e-3a8a00088810">

</details>

As we can see, the alert was generated but the rule was not updated.

# Task
- [ ] Research the `gcp_posture` rules and confirm that the [change suggested](https://documentation-dev.wazuh.com/v4.9.0-alpha3/cloud-security/gcp/posture-management.html#network-misconfigurations) should trigger a MISCONFIGURATION event.
- [ ]  Update the documentation accordingly.

**Comment by ooniagbi** [comment]
We'll investigate and update the issue. However, after an initial review, I observed that the misconfiguration event is either not being generated or not being collected by Wazuh. If either is the case, then it is not a rule issue.

**Comment by Johnng007** [comment]
Environment has been setup and testing has began.

**Comment by Johnng007** [comment]
<details>
<summary>Add new firewall rule</summary>

the field name gcp.finding.findingClass is no longer included in the GCP alert as seen from the new firewall creation alert below.

```bash
{
  "_index": "wazuh-alerts-4.x-2024.07.29",
  "_id": "jM8AAJEBujQaUbddq4UJ",
  "_version": 1,
  "_score": null,
  "_source": {
    "input": {
      "type": "log"
    },
    "agent": {
      "name": "stockholm-virtual-machine",
      "id": "000"
    },
    "manager": {
      "name": "stockholm-virtual-machine"
    },
    "data": {
      "gcp": {
        "severity": "NOTICE",
        "logName": "projects/wazuh-430915/logs/cloudaudit.googleapis.com%2Factivity",
        "resource": {
          "type": "gce_firewall_rule",
          "labels": {
            "project_id": "wazuh-430915",
            "firewall_rule_id": "7407844747802988942"
          }
        },
        "protoPayload": {
          "requestMetadata": {
            "requestAttributes": {
              "reason": "8uSywAYQGg5Db2xpc2V1bSBGbG93cw",
              "time": "2024-07-29T19:37:05.34935Z"
            },
            "callerSuppliedUserAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36,gzip(gfe),gzip(gfe)",
            "callerIp": "93.153.23.240"
          },
          "request": {
            "logConfig": {
              "enable": "false"
            },
            "sourceRanges": [
              "0.0.0.0/0"
            ],
            "@type": "type.googleapis.com/compute.firewalls.insert",
            "name": "verybadrule",
            "alloweds": [
              {
                "IPProtocol": "all"
              }
            ],
            "disabled": "false",
            "priority": "1000",
            "direction": "INGRESS",
            "network": "projects/wazuh-430915/global/networks/default"
          },
          "authenticationInfo": {
            "principalEmail": "ony***@gmail.com"
          },
          "authorizationInfo": [
            {
              "permissionType": "ADMIN_WRITE",
              "resource": "projects/wazuh-430915/global/firewalls/verybadrule",
              "permission": "compute.firewalls.create",
              "resourceAttributes": {
                "service": "compute",
                "name": "projects/wazuh-430915/global/firewalls/verybadrule",
                "type": "compute.firewalls"
              },
              "granted": true
            },
            {
              "permissionType": "ADMIN_WRITE",
              "resource": "projects/wazuh-430915/global/networks/default",
              "permission": "compute.networks.updatePolicy",
              "resourceAttributes": {
                "service": "compute",
                "name": "projects/wazuh-430915/global/networks/default",
                "type": "compute.networks"
              },
              "granted": true
            }
          ],
          "@type": "type.googleapis.com/google.cloud.audit.AuditLog",
          "response": {
            "targetId": "7407844747802988942",
            "@type": "type.googleapis.com/operation",
            "selfLink": "https://www.googleapis.com/compute/beta/projects/wazuh-430915/global/operations/operation-1722281824871-61e67fc5fe439-84be46b0-472e9b6f",
            "insertTime": "2024-07-29T12:37:05.232-07:00",
            "selfLinkWithId": "https://www.googleapis.com/compute/beta/projects/wazuh-430915/global/operations/7688623231142089102",
            "targetLink": "https://www.googleapis.com/compute/beta/projects/wazuh-430915/global/firewalls/verybadrule",
            "name": "operation-1722281824871-61e67fc5fe439-84be46b0-472e9b6f",
            "progress": "0",
            "operationType": "insert",
            "startTime": "2024-07-29T12:37:05.240-07:00",
            "id": "7688623231142089102",
            "user": "onyejegbujohne@gmail.com",
            "status": "RUNNING"
          },
          "methodName": "beta.compute.firewalls.insert",
          "resourceName": "projects/wazuh-430915/global/firewalls/verybadrule",
          "resourceLocation": {
            "currentLocations": [
              "global"
            ]
          },
          "serviceName": "compute.googleapis.com"
        },
        "receiveTimestamp": "2024-07-29T19:37:06.299705555Z",
        "operation": {
          "producer": "compute.googleapis.com",
          "id": "operation-1722281824871-61e67fc5fe439-84be46b0-472e9b6f",
          "first": "true"
        },
        "insertId": "-6x8uoge2b5is",
        "labels": {
          "compute": {
            "googleapis": {
              "com/root_trigger_id": "adbedd39-ab4e-4a21-b3ed-8504be159c47"
            }
          }
        },
        "timestamp": "2024-07-29T19:37:05.002481Z"
      },
      "integration": "gcp"
    },
    "rule": {
      "firedtimes": 63,
      "mail": false,
      "level": 3,
      "description": "GCP notice event on project wazuh-430915, monitored resource type: gce_firewall_rule.",
      "groups": [
        "gcp"
      ],
      "id": "65042"
    },
    "location": "Wazuh-GCloud",
    "decoder": {
      "name": "json"
    },
    "id": "1722281927.1034411",
    "timestamp": "2024-07-29T21:38:47.781+0200"
  },
  "fields": {
    "timestamp": [
      "2024-07-29T19:38:47.781Z"
    ]
  },
  "sort": [
    1722281927781
  ]
}
```
</details>

The rule heavily depends on the field name gcp.finding.findingClass to trigger. 
But this field is now exclusively an SCC (Security Command Center) feature.
https://cloud.google.com/security-command-center/docs/finding-classes

what is included in the alerts is a severity rating that is not influenced by any intelligence, as a misconfigured firewall is still given a `NOTICE` severity rating.

![image](https://github.com/user-attachments/assets/c2216abd-f3e1-4d5a-92dc-3c2ad2f268b5)

**Comment by Johnng007** [comment]
Putting this issue on hold, while i request for a GCP subscription access.

**Comment by Johnng007** [comment]
I have repeated this test using the resources and credentials supplied by the devel team.

A user outside the organization with IAM roles was triggered as a Misconfiguration alert as shown below.

<details>
<summary>I AM User </summary>

```
{
  "_index": "wazuh-alerts-4.x-2024.07.31",
  "_id": "NjAhCpEBzORuATO-FKfN",
  "_version": 1,
  "_score": null,
  "_source": {
    "input": {
      "type": "log"
    },
    "agent": {
      "name": "wazuh-server",
      "id": "000"
    },
    "manager": {
      "name": "wazuh-server"
    },
    "data": {
      "gcp": {
        "resource": {
          "resourcePathString": "organizations/101551879186/projects/769054035614",
          "parent": "//cloudresourcemanager.googleapis.com/organizations/101551879186",
          "projectDisplayName": "wazuh-dev-258815",
          "displayName": "wazuh-dev-258815",
          "service": "cloudresourcemanager.googleapis.com",
          "cloudProvider": "GOOGLE_CLOUD_PLATFORM",
          "organization": "organizations/101551879186",
          "resourcePath": {
            "nodes": [
              {
                "displayName": "wazuh-dev-258815",
                "id": "projects/769054035614",
                "nodeType": "GCP_PROJECT"
              },
              {
                "id": "organizations/101551879186",
                "nodeType": "GCP_ORGANIZATION"
              }
            ]
          },
          "name": "//cloudresourcemanager.googleapis.com/projects/769054035614",
          "project": "//cloudresourcemanager.googleapis.com/projects/769054035614",
          "type": "google.cloud.resourcemanager.Project",
          "parentDisplayName": "wazuh.com"
        },
        "notificationConfigName": "projects/769054035614/notificationConfigs/GCPFindings",
        "finding": {
          "severity": "HIGH",
          "parent": "organizations/101551879186/sources/6049642661091069920",
          "sourceProperties": {
            "ResourcePath": [
              "projects/wazuh-dev-258815/",
              "organizations/101551879186/"
            ],
            "ReactivationCount": "0",
            "Explanation": "A user outside of your organization has IAM permissions on a project, folder, or organization.",
            "ScannerName": "IAM_SCANNER",
            "OffendingIamRoles": "{\"invalidRoles\":[{\"user\":\"user:johnon***u@gmail.com\",\"roles\":[\"roles/resourcemanager.projectOwnerInvitee\"]}]}",
            "compliance_standards": {
              "iso": [
                {
                  "ids": [
                    "A.9.2.3"
                  ]
                }
              ],
              "pci": [
                {
                  "ids": [
                    "7.1.2"
                  ]
                }
              ],
              "cis": [
                {
                  "ids": [
                    "1.1",
                    [
                      "1.1"
                    ],
                    [
                      "1.1"
                    ],
                    [
                      "1.1"
                    ],
                    [
                      "1.1"
                    ]
                  ],
                  "version": "1.0"
                },
                {
                  "ids": [
                    "1.1"
                  ],
                  "version": "1.1"
                },
                {
                  "ids": [
                    "1.1"
                  ],
                  "version": "1.2"
                },
                {
                  "ids": [
                    "1.1"
                  ],
                  "version": "1.3"
                },
                {
                  "ids": [
                    "1.1"
                  ],
                  "version": "2.0"
                }
              ],
              "nist": [
                {
                  "ids": [
                    "AC-3"
                  ]
                }
              ]
            },
            "ExceptionInstructions": "Add the security mark \"allow_non_org_iam_member\" to the asset with a value of \"true\" to prevent this finding from being activated again.",
            "OffendingIamRolesList": [
              {
                "roles": [
                  "roles/resourcemanager.projectOwnerInvitee"
                ],
                "member": "user:johno***u@gmail.com"
              }
            ],
            "Recommendation": "Go to https://console.cloud.google.com/iam-admin/iam?project=wazuh-dev-258815 and remove entries for users which are not in your organization (e.g. gmail.com addresses). Invalid accounts: [user:johno***u@gmail.com]",
            "cli_remediation": [
              "gcloud projects remove-iam-policy-binding wazuh-dev-258815 --member=user:johno***u@gmail.com --role=roles/resourcemanager.projectOwnerInvitee"
            ]
          },
          "findingClass": "MISCONFIGURATION",
          "externalUri": "https://console.cloud.google.com/iam-admin/iam?project=wazuh-dev-***",
          "iamBindings": [
            {
              "role": "roles/resourcemanager.projectOwnerInvitee",
              "member": "user:johno***u@gmail.com"
            }
          ],
          "securityMarks": {
            "name": "organizations/101551879186/sources/6049642661091069920/findings/93688b376eaa00da90764505af2a0e12/securityMarks"
          },
          "description": "A user outside of your organization has IAM permissions on a project, folder, or organization.",
          "resourceName": "//cloudresourcemanager.googleapis.com/projects/769054035614",
          "mute": "UNDEFINED",
          "parentDisplayName": "Security Health Analytics",
          "compliances": [
            {
              "standard": "cis",
              "ids": [
                "1.1",
                [
                  "1.1"
                ],
                [
                  "1.1"
                ],
                [
                  "1.1"
                ],
                [
                  "1.1"
                ],
                [
                  "AC-3"
                ],
                [
                  "7.1.2"
                ],
                [
                  "A.9.2.3"
                ]
              ],
              "version": "1.0"
            },
            {
              "standard": "cis",
              "ids": [
                "1.1"
              ],
              "version": "1.1"
            },
            {
              "standard": "cis",
              "ids": [
                "1.1"
              ],
              "version": "1.2"
            },
            {
              "standard": "cis",
              "ids": [
                "1.1"
              ],
              "version": "1.3"
            },
            {
              "standard": "cis",
              "ids": [
                "1.1"
              ],
              "version": "2.0"
            },
            {
              "standard": "nist",
              "ids": [
                "AC-3"
              ]
            },
            {
              "standard": "pci",
              "ids": [
                "7.1.2"
              ]
            },
            {
              "standard": "iso",
              "ids": [
                "A.9.2.3"
              ]
            }
          ],
          "createTime": "2024-07-31T18:33:16.175Z",
          "muteUpdateTime": "1970-01-01T00:00:00Z",
          "name": "organizations/101551879186/sources/6049642661091069920/findings/93688b376eaa00da90764505af2a0e12",
          "eventTime": "2024-07-31T18:33:15.818442Z",
          "state": "ACTIVE",
          "category": "NON_ORG_IAM_MEMBER",
          "canonicalName": "projects/769054035614/sources/6049642661091069920/findings/93688b376eaa00da90764505af2a0e12"
        }
      },
      "integration": "gcp"
    },
    "rule": {
      "firedtimes": 1,
      "mail": false,
      "level": 10,
      "description": "A MISCONFIGURATION with HIGH severity has been discovered on the GCP project wazuh-dev-258815. A user outside of your organization has IAM permissions on a project, folder, or organization.",
      "groups": [
        "gcp"
      ],
      "mitre": {
        "technique": [
          "Impair Defenses"
        ],
        "id": [
          "T1562"
        ],
        "tactic": [
          "Defense Evasion"
        ]
      },
      "id": "100200"
    },
    "location": "Wazuh-GCloud",
    "decoder": {
      "name": "json"
    },
    "id": "1722451822.929814",
    "full_log": "{\"integration\": \"gcp\", \"gcp\": {\n  \"notificationConfigName\": \"projects/769054035614/notificationConfigs/GCPFindings\",\n  \"finding\": {\n    \"name\": \"organizations/101551879186/sources/6049642661091069920/findings/93688b376eaa00da90764505af2a0e12\",\n    \"parent\": \"organizations/101551879186/sources/6049642661091069920\",\n    \"resourceName\": \"//cloudresourcemanager.googleapis.com/projects/769054035614\",\n    \"state\": \"ACTIVE\",\n    \"category\": \"NON_ORG_IAM_MEMBER\",\n    \"externalUri\": \"https://console.cloud.google.com/iam-admin/iam?project\\u003dwazuh-dev-258815\",\n    \"sourceProperties\": {\n      \"Recommendation\": \"Go to https://console.cloud.google.com/iam-admin/iam?project\\u003dwazuh-dev-258815 and remove entries for users which are not in your organization (e.g. gmail.com addresses). Invalid accounts: [user:johnonyejegbu@gmail.com]\",\n      \"ExceptionInstructions\": \"Add the security mark \\\"allow_non_org_iam_member\\\" to the asset with a value of \\\"true\\\" to prevent this finding from being activated again.\",\n      \"Explanation\": \"A user outside of your organization has IAM permissions on a project, folder, or organization.\",\n      \"ScannerName\": \"IAM_SCANNER\",\n      \"ResourcePath\": [\"projects/wazuh-dev-258815/\", \"organizations/101551879186/\"],\n      \"cli_remediation\": [\"gcloud projects remove-iam-policy-binding wazuh-dev-258815 --member\\u003duser:johnonyejegbu@gmail.com --role\\u003droles/resourcemanager.projectOwnerInvitee\"],\n      \"OffendingIamRoles\": \"{\\\"invalidRoles\\\":[{\\\"user\\\":\\\"user:johnonyejegbu@gmail.com\\\",\\\"roles\\\":[\\\"roles/resourcemanager.projectOwnerInvitee\\\"]}]}\",\n      \"OffendingIamRolesList\": [{\n        \"member\": \"user:johnonyejegbu@gmail.com\",\n        \"roles\": [\"roles/resourcemanager.projectOwnerInvitee\"]\n      }],\n      \"compliance_standards\": {\n        \"iso\": [{\n          \"ids\": [\"A.9.2.3\"]\n        }],\n        \"pci\": [{\n          \"ids\": [\"7.1.2\"]\n        }],\n        \"cis\": [{\n          \"version\": \"1.0\",\n          \"ids\": [\"1.1\"]\n        }, {\n          \"version\": \"1.1\",\n          \"ids\": [\"1.1\"]\n        }, {\n          \"version\": \"1.2\",\n          \"ids\": [\"1.1\"]\n        }, {\n          \"version\": \"1.3\",\n          \"ids\": [\"1.1\"]\n        }, {\n          \"version\": \"2.0\",\n          \"ids\": [\"1.1\"]\n        }],\n        \"nist\": [{\n          \"ids\": [\"AC-3\"]\n        }]\n      },\n      \"ReactivationCount\": 0.0\n    },\n    \"securityMarks\": {\n      \"name\": \"organizations/101551879186/sources/6049642661091069920/findings/93688b376eaa00da90764505af2a0e12/securityMarks\"\n    },\n    \"eventTime\": \"2024-07-31T18:33:15.818442Z\",\n    \"createTime\": \"2024-07-31T18:33:16.175Z\",\n    \"severity\": \"HIGH\",\n    \"canonicalName\": \"projects/769054035614/sources/6049642661091069920/findings/93688b376eaa00da90764505af2a0e12\",\n    \"mute\": \"UNDEFINED\",\n    \"findingClass\": \"MISCONFIGURATION\",\n    \"muteUpdateTime\": \"1970-01-01T00:00:00Z\",\n    \"compliances\": [{\n      \"standard\": \"cis\",\n      \"version\": \"1.0\",\n      \"ids\": [\"1.1\"]\n    }, {\n      \"standard\": \"cis\",\n      \"version\": \"1.1\",\n      \"ids\": [\"1.1\"]\n    }, {\n      \"standard\": \"cis\",\n      \"version\": \"1.2\",\n      \"ids\": [\"1.1\"]\n    }, {\n      \"standard\": \"cis\",\n      \"version\": \"1.3\",\n      \"ids\": [\"1.1\"]\n    }, {\n      \"standard\": \"cis\",\n      \"version\": \"2.0\",\n      \"ids\": [\"1.1\"]\n    }, {\n      \"standard\": \"nist\",\n      \"ids\": [\"AC-3\"]\n    }, {\n      \"standard\": \"pci\",\n      \"ids\": [\"7.1.2\"]\n    }, {\n      \"standard\": \"iso\",\n      \"ids\": [\"A.9.2.3\"]\n    }],\n    \"parentDisplayName\": \"Security Health Analytics\",\n    \"description\": \"A user outside of your organization has IAM permissions on a project, folder, or organization.\",\n    \"iamBindings\": [{\n      \"role\": \"roles/resourcemanager.projectOwnerInvitee\",\n      \"member\": \"user:johnonyejegbu@gmail.com\"\n    }]\n  },\n  \"resource\": {\n    \"name\": \"//cloudresourcemanager.googleapis.com/projects/769054035614\",\n    \"project\": \"//cloudresourcemanager.googleapis.com/projects/769054035614\",\n    \"projectDisplayName\": \"wazuh-dev-258815\",\n    \"parent\": \"//cloudresourcemanager.googleapis.com/organizations/101551879186\",\n    \"parentDisplayName\": \"wazuh.com\",\n    \"type\": \"google.cloud.resourcemanager.Project\",\n    \"displayName\": \"wazuh-dev-258815\",\n    \"cloudProvider\": \"GOOGLE_CLOUD_PLATFORM\",\n    \"organization\": \"organizations/101551879186\",\n    \"service\": \"cloudresourcemanager.googleapis.com\",\n    \"resourcePath\": {\n      \"nodes\": [{\n        \"nodeType\": \"GCP_PROJECT\",\n        \"id\": \"projects/769054035614\",\n        \"displayName\": \"wazuh-dev-258815\"\n      }, {\n        \"nodeType\": \"GCP_ORGANIZATION\",\n        \"id\": \"organizations/101551879186\"\n      }]\n    },\n    \"resourcePathString\": \"organizations/101551879186/projects/769054035614\"\n  }\n}}",
    "timestamp": "2024-07-31T18:50:22.945+0000"
  },
  "fields": {
    "timestamp": [
      "2024-07-31T18:50:22.945Z"
    ]
  },
  "sort": [
    1722451822945
  ]
}
```

![image](https://github.com/user-attachments/assets/cb060c93-209b-4daa-9748-753734a28e02)

</details>

**Comment by Johnng007** [comment]
I have repeated this test using the resources and credentials supplied by the devel team.

A user outside the organization with IAM roles was triggered as a Misconfiguration alert as shown below.

<details>
<summary>I AM User </summary>

```
{
  "_index": "wazuh-alerts-4.x-2024.07.31",
  "_id": "NjAhCpEBzORuATO-FKfN",
  "_version": 1,
  "_score": null,
  "_source": {
    "input": {
      "type": "log"
    },
    "agent": {
      "name": "wazuh-server",
      "id": "000"
    },
    "manager": {
      "name": "wazuh-server"
    },
    "data": {
      "gcp": {
        "resource": {
          "resourcePathString": "organizations/101551879186/projects/769054035614",
          "parent": "//cloudresourcemanager.googleapis.com/organizations/101551879186",
          "projectDisplayName": "wazuh-dev-258815",
          "displayName": "wazuh-dev-258815",
          "service": "cloudresourcemanager.googleapis.com",
          "cloudProvider": "GOOGLE_CLOUD_PLATFORM",
          "organization": "organizations/101551879186",
          "resourcePath": {
            "nodes": [
              {
                "displayName": "wazuh-dev-258815",
                "id": "projects/769054035614",
                "nodeType": "GCP_PROJECT"
              },
              {
                "id": "organizations/101551879186",
                "nodeType": "GCP_ORGANIZATION"
              }
            ]
          },
          "name": "//cloudresourcemanager.googleapis.com/projects/769054035614",
          "project": "//cloudresourcemanager.googleapis.com/projects/769054035614",
          "type": "google.cloud.resourcemanager.Project",
          "parentDisplayName": "wazuh.com"
        },
        "notificationConfigName": "projects/769054035614/notificationConfigs/GCPFindings",
        "finding": {
          "severity": "HIGH",
          "parent": "organizations/101551879186/sources/6049642661091069920",
          "sourceProperties": {
            "ResourcePath": [
              "projects/wazuh-dev-258815/",
              "organizations/101551879186/"
            ],
            "ReactivationCount": "0",
            "Explanation": "A user outside of your organization has IAM permissions on a project, folder, or organization.",
            "ScannerName": "IAM_SCANNER",
            "OffendingIamRoles": "{\"invalidRoles\":[{\"user\":\"user:johnon***u@gmail.com\",\"roles\":[\"roles/resourcemanager.projectOwnerInvitee\"]}]}",
            "compliance_standards": {
              "iso": [
                {
                  "ids": [
                    "A.9.2.3"
                  ]
                }
              ],
              "pci": [
                {
                  "ids": [
                    "7.1.2"
                  ]
                }
              ],
              "cis": [
                {
                  "ids": [
                    "1.1",
                    [
                      "1.1"
                    ],
                    [
                      "1.1"
                    ],
                    [
                      "1.1"
                    ],
                    [
                      "1.1"
                    ]
                  ],
                  "version": "1.0"
                },
                {
                  "ids": [
                    "1.1"
                  ],
                  "version": "1.1"
                },
                {
                  "ids": [
                    "1.1"
                  ],
                  "version": "1.2"
                },
                {
                  "ids": [
                    "1.1"
                  ],
                  "version": "1.3"
                },
                {
                  "ids": [
                    "1.1"
                  ],
                  "version": "2.0"
                }
              ],
              "nist": [
                {
                  "ids": [
                    "AC-3"
                  ]
                }
              ]
            },
            "ExceptionInstructions": "Add the security mark \"allow_non_org_iam_member\" to the asset with a value of \"true\" to prevent this finding from being activated again.",
            "OffendingIamRolesList": [
              {
                "roles": [
                  "roles/resourcemanager.projectOwnerInvitee"
                ],
                "member": "user:johno***u@gmail.com"
              }
            ],
            "Recommendation": "Go to https://console.cloud.google.com/iam-admin/iam?project=wazuh-dev-258815 and remove entries for users which are not in your organization (e.g. gmail.com addresses). Invalid accounts: [user:johno***u@gmail.com]",
            "cli_remediation": [
              "gcloud projects remove-iam-policy-binding wazuh-dev-258815 --member=user:johno***u@gmail.com --role=roles/resourcemanager.projectOwnerInvitee"
            ]
          },
          "findingClass": "MISCONFIGURATION",
          "externalUri": "https://console.cloud.google.com/iam-admin/iam?project=wazuh-dev-***",
          "iamBindings": [
            {
              "role": "roles/resourcemanager.projectOwnerInvitee",
              "member": "user:johno***u@gmail.com"
            }
          ],
          "securityMarks": {
            "name": "organizations/101551879186/sources/6049642661091069920/findings/93688b376eaa00da90764505af2a0e12/securityMarks"
          },
          "description": "A user outside of your organization has IAM permissions on a project, folder, or organization.",
          "resourceName": "//cloudresourcemanager.googleapis.com/projects/769054035614",
          "mute": "UNDEFINED",
          "parentDisplayName": "Security Health Analytics",
          "compliances": [
            {
              "standard": "cis",
              "ids": [
                "1.1",
                [
                  "1.1"
                ],
                [
                  "1.1"
                ],
                [
                  "1.1"
                ],
                [
                  "1.1"
                ],
                [
                  "AC-3"
                ],
                [
                  "7.1.2"
                ],
                [
                  "A.9.2.3"
                ]
              ],
              "version": "1.0"
            },
            {
              "standard": "cis",
              "ids": [
                "1.1"
              ],
              "version": "1.1"
            },
            {
              "standard": "cis",
              "ids": [
                "1.1"
              ],
              "version": "1.2"
            },
            {
              "standard": "cis",
              "ids": [
                "1.1"
              ],
              "version": "1.3"
            },
            {
              "standard": "cis",
              "ids": [
                "1.1"
              ],
              "version": "2.0"
            },
            {
              "standard": "nist",
              "ids": [
                "AC-3"
              ]
            },
            {
              "standard": "pci",
              "ids": [
                "7.1.2"
              ]
            },
            {
              "standard": "iso",
              "ids": [
                "A.9.2.3"
              ]
            }
          ],
          "createTime": "2024-07-31T18:33:16.175Z",
          "muteUpdateTime": "1970-01-01T00:00:00Z",
          "name": "organizations/101551879186/sources/6049642661091069920/findings/93688b376eaa00da90764505af2a0e12",
          "eventTime": "2024-07-31T18:33:15.818442Z",
          "state": "ACTIVE",
          "category": "NON_ORG_IAM_MEMBER",
          "canonicalName": "projects/769054035614/sources/6049642661091069920/findings/93688b376eaa00da90764505af2a0e12"
        }
      },
      "integration": "gcp"
    },
    "rule": {
      "firedtimes": 1,
      "mail": false,
      "level": 10,
      "description": "A MISCONFIGURATION with HIGH severity has been discovered on the GCP project wazuh-dev-258815. A user outside of your organization has IAM permissions on a project, folder, or organization.",
      "groups": [
        "gcp"
      ],
      "mitre": {
        "technique": [
          "Impair Defenses"
        ],
        "id": [
          "T1562"
        ],
        "tactic": [
          "Defense Evasion"
        ]
      },
      "id": "100200"
    },
    "location": "Wazuh-GCloud",
    "decoder": {
      "name": "json"
    },
    "id": "1722451822.929814",
    "full_log": "{\"integration\": \"gcp\", \"gcp\": {\n  \"notificationConfigName\": \"projects/769054035614/notificationConfigs/GCPFindings\",\n  \"finding\": {\n    \"name\": \"organizations/101551879186/sources/6049642661091069920/findings/93688b376eaa00da90764505af2a0e12\",\n    \"parent\": \"organizations/101551879186/sources/6049642661091069920\",\n    \"resourceName\": \"//cloudresourcemanager.googleapis.com/projects/769054035614\",\n    \"state\": \"ACTIVE\",\n    \"category\": \"NON_ORG_IAM_MEMBER\",\n    \"externalUri\": \"https://console.cloud.google.com/iam-admin/iam?project\\u003dwazuh-dev-258815\",\n    \"sourceProperties\": {\n      \"Recommendation\": \"Go to https://console.cloud.google.com/iam-admin/iam?project\\u003dwazuh-dev-258815 and remove entries for users which are not in your organization (e.g. gmail.com addresses). Invalid accounts: [user:johnonyejegbu@gmail.com]\",\n      \"ExceptionInstructions\": \"Add the security mark \\\"allow_non_org_iam_member\\\" to the asset with a value of \\\"true\\\" to prevent this finding from being activated again.\",\n      \"Explanation\": \"A user outside of your organization has IAM permissions on a project, folder, or organization.\",\n      \"ScannerName\": \"IAM_SCANNER\",\n      \"ResourcePath\": [\"projects/wazuh-dev-258815/\", \"organizations/101551879186/\"],\n      \"cli_remediation\": [\"gcloud projects remove-iam-policy-binding wazuh-dev-258815 --member\\u003duser:johnonyejegbu@gmail.com --role\\u003droles/resourcemanager.projectOwnerInvitee\"],\n      \"OffendingIamRoles\": \"{\\\"invalidRoles\\\":[{\\\"user\\\":\\\"user:johnonyejegbu@gmail.com\\\",\\\"roles\\\":[\\\"roles/resourcemanager.projectOwnerInvitee\\\"]}]}\",\n      \"OffendingIamRolesList\": [{\n        \"member\": \"user:johnonyejegbu@gmail.com\",\n        \"roles\": [\"roles/resourcemanager.projectOwnerInvitee\"]\n      }],\n      \"compliance_standards\": {\n        \"iso\": [{\n          \"ids\": [\"A.9.2.3\"]\n        }],\n        \"pci\": [{\n          \"ids\": [\"7.1.2\"]\n        }],\n        \"cis\": [{\n          \"version\": \"1.0\",\n          \"ids\": [\"1.1\"]\n        }, {\n          \"version\": \"1.1\",\n          \"ids\": [\"1.1\"]\n        }, {\n          \"version\": \"1.2\",\n          \"ids\": [\"1.1\"]\n        }, {\n          \"version\": \"1.3\",\n          \"ids\": [\"1.1\"]\n        }, {\n          \"version\": \"2.0\",\n          \"ids\": [\"1.1\"]\n        }],\n        \"nist\": [{\n          \"ids\": [\"AC-3\"]\n        }]\n      },\n      \"ReactivationCount\": 0.0\n    },\n    \"securityMarks\": {\n      \"name\": \"organizations/101551879186/sources/6049642661091069920/findings/93688b376eaa00da90764505af2a0e12/securityMarks\"\n    },\n    \"eventTime\": \"2024-07-31T18:33:15.818442Z\",\n    \"createTime\": \"2024-07-31T18:33:16.175Z\",\n    \"severity\": \"HIGH\",\n    \"canonicalName\": \"projects/769054035614/sources/6049642661091069920/findings/93688b376eaa00da90764505af2a0e12\",\n    \"mute\": \"UNDEFINED\",\n    \"findingClass\": \"MISCONFIGURATION\",\n    \"muteUpdateTime\": \"1970-01-01T00:00:00Z\",\n    \"compliances\": [{\n      \"standard\": \"cis\",\n      \"version\": \"1.0\",\n      \"ids\": [\"1.1\"]\n    }, {\n      \"standard\": \"cis\",\n      \"version\": \"1.1\",\n      \"ids\": [\"1.1\"]\n    }, {\n      \"standard\": \"cis\",\n      \"version\": \"1.2\",\n      \"ids\": [\"1.1\"]\n    }, {\n      \"standard\": \"cis\",\n      \"version\": \"1.3\",\n      \"ids\": [\"1.1\"]\n    }, {\n      \"standard\": \"cis\",\n      \"version\": \"2.0\",\n      \"ids\": [\"1.1\"]\n    }, {\n      \"standard\": \"nist\",\n      \"ids\": [\"AC-3\"]\n    }, {\n      \"standard\": \"pci\",\n      \"ids\": [\"7.1.2\"]\n    }, {\n      \"standard\": \"iso\",\n      \"ids\": [\"A.9.2.3\"]\n    }],\n    \"parentDisplayName\": \"Security Health Analytics\",\n    \"description\": \"A user outside of your organization has IAM permissions on a project, folder, or organization.\",\n    \"iamBindings\": [{\n      \"role\": \"roles/resourcemanager.projectOwnerInvitee\",\n      \"member\": \"user:johnonyejegbu@gmail.com\"\n    }]\n  },\n  \"resource\": {\n    \"name\": \"//cloudresourcemanager.googleapis.com/projects/769054035614\",\n    \"project\": \"//cloudresourcemanager.googleapis.com/projects/769054035614\",\n    \"projectDisplayName\": \"wazuh-dev-258815\",\n    \"parent\": \"//cloudresourcemanager.googleapis.com/organizations/101551879186\",\n    \"parentDisplayName\": \"wazuh.com\",\n    \"type\": \"google.cloud.resourcemanager.Project\",\n    \"displayName\": \"wazuh-dev-258815\",\n    \"cloudProvider\": \"GOOGLE_CLOUD_PLATFORM\",\n    \"organization\": \"organizations/101551879186\",\n    \"service\": \"cloudresourcemanager.googleapis.com\",\n    \"resourcePath\": {\n      \"nodes\": [{\n        \"nodeType\": \"GCP_PROJECT\",\n        \"id\": \"projects/769054035614\",\n        \"displayName\": \"wazuh-dev-258815\"\n      }, {\n        \"nodeType\": \"GCP_ORGANIZATION\",\n        \"id\": \"organizations/101551879186\"\n      }]\n    },\n    \"resourcePathString\": \"organizations/101551879186/projects/769054035614\"\n  }\n}}",
    "timestamp": "2024-07-31T18:50:22.945+0000"
  },
  "fields": {
    "timestamp": [
      "2024-07-31T18:50:22.945Z"
    ]
  },
  "sort": [
    1722451822945
  ]
}
```

![image](https://github.com/user-attachments/assets/cb060c93-209b-4daa-9748-753734a28e02)

</details>

**Comment by ooniagbi** [comment]
What about the firewall misconfiguration event?

**Comment by Johnng007** [comment]
> What about the firewall misconfiguration event?

Firewall Misconfiguration events were still coming in as previously noted in my first analysis.
But i still have some firewall permission Errors. Working with the devops team to fix this.

**Comment by Johnng007** [comment]
From observations so far, we have noticed that created misconfigured firewall rules are not being detected in the Google Security Command Center, and we depend on this detection for our rule trigger.
![image](https://github.com/user-attachments/assets/db8d942e-d372-48f8-90b3-a685dea219d8)

Until we are able to figure out why these detections are not happening in GSCC  i will set this issue to on-hold.

**Comment by IsExec** [comment]
# Issue Update

Aim: Confirm if bad firewall rule is detected as misconfiguration in GCP

- Wazuh manager environment setup
- GCP personal account setup

### Result: misconfigured firewall rules are not being detected in the Google Security Command Center.

![image](https://github.com/user-attachments/assets/0358db37-afdb-4625-92ad-e23d55a9a913)

## Next Step

- RCA for why firewall rule didn't trigger in Google Security Command Center

**Comment by IsExec** [comment]
# Issue Update

The rules worked as intended and the alerts were triggered correctly.

<details>
<summary>Findings in GCP</summary>

![image](https://github.com/user-attachments/assets/5e884b1a-e569-43fc-a1c5-e4f173a4dc7b)

</details>

<details>
<summary>Wazuh Dashboard</summary>

![image](https://github.com/user-attachments/assets/9c93a91f-d9bc-437a-acc7-50902304d11f)

</details>

## Observation
- It took time (hours) for the 'verybadrule' misconfiguration to trigger alert on GCP security command center Finding **for the first time**, once it did, it became consistent.
- This was also confirmed with @Johnng007 project which took more than 24 hours to trigger the first alert on GCP security command center Finding.
- Once the alerts were triggered on GCP security command center Finding, the events and corresponding alert were found on Wazuh Dashboard.
- Further research shows that upon activating the security command center for the first, there is an activation process that must be completed before the resources are scanned and the scan must be completed before complete set of findings are displayed as reported in Google Documentation - [Security Command Center latency overview](https://cloud.google.com/security-command-center/docs/concepts-scan-latency-overview).
- This latency experienced varies from user to user and detection modules based on a number of factors such as number of assets and resources. 

## Conclusion
The latency experienced for the first time before findings that can trigger the gcp_posture.xml rules in Wazuh is the regular working of the GCP security command center.

## Recommendation
The [documentation](https://wazuh.com/blog/cloud-security-posture-management/) should be updated to reflect the latency as explained in the Google Documentation to prepare the users and provide comfort that this is not a problem with Wazuh dashboard neither is the user doing the wrong thing.

## Reference
Google - https://cloud.google.com/security-command-center/docs/concepts-scan-latency-overview

**Comment by ooniagbi** [comment]
Documentation updated. Closing as there are no pending tasks.
